### PR TITLE
Added style block validation

### LINKED ISSUE #1104: Block errors on upgrade
When upgraded to v2.14.1, these blocks produced block errors:
feature grid, count up, button, icon list, number box, feature, spacer

here's the link to the code of the blocks:
https://gist.github.com/legaspielaine/0cc76b17dcc9777053120d5108ef527b

### PR DESCRIPTION
**Pull request by blankart**
Fixes #1104

### CODE REVIEW

File: src/plugins/auto-block-recovery/is-invalid.js [hunk 1]
```diff
@@ -618,3 +556,36 @@ export const isFocusable = issue => {
...
+	}
+
+	let expectedContent
+	try {
+		expectedContent = getSaveContent( name, attributes, innerBlocks )
+	} catch ( error ) {
+		return false
+	}
+
+	const originalStyleContent = originalContent.match( /<style>[^\<]*<\/style>/g )?.[ 0 ]
```
Comment (bfintal) [inline src/plugins/auto-block-recovery/is-invalid.js:583]: This can fail if you have an inline svg as a css value.

Comment (blankart) [inline src/plugins/auto-block-recovery/is-invalid.js:583]:
```suggestion
	const originalStyleContent = originalContent.match( /<style>(.+?(?=<\/style>))*<\/style>/g )?.[ 0 ]
```

File: src/plugins/auto-block-recovery/is-invalid.js [hunk 2]
```diff
@@ -618,3 +556,36 @@ export const isFocusable = issue => {
...
+
+	let expectedContent
+	try {
+		expectedContent = getSaveContent( name, attributes, innerBlocks )
+	} catch ( error ) {
+		return false
+	}
+
+	const originalStyleContent = originalContent.match( /<style>(.+?(?=<\/style>))*<\/style>/g )?.[ 0 ]
+	const expectedStyleContent = expectedContent.match( /<style>[^\<]*<\/style>/g )?.[ 0 ]
```
Comment (blankart) [inline src/plugins/auto-block-recovery/is-invalid.js:584]:
```suggestion
	const expectedStyleContent = expectedContent.match( /<style>(.+?(?=<\/style>))*<\/style>/g )?.[ 0 ]
```

File: src/plugins/auto-block-recovery/is-invalid.js [hunk 3]
```diff
@@ -618,3 +556,36 @@ export const isFocusable = issue => {
...
+
+	let expectedContent
+	try {
+		expectedContent = getSaveContent( name, attributes, innerBlocks )
+	} catch ( error ) {
+		return false
+	}
+
+	const originalStyleContent = originalContent.match( /<style>(.+?(?=<\/style>))*<\/style>/g )?.[ 0 ]
+	const expectedStyleContent = expectedContent.match( /<style>(.+?(?=<\/style>))*<\/style>/g )?.[ 0 ]
```
Comment (blankart) [inline src/plugins/auto-block-recovery/is-invalid.js:584]:
```suggestion
	const originalStyleContent = originalContent.match( /<style>(.*)?<\/style>/g )?.[ 0 ]
	const expectedStyleContent = expectedContent.match( /<style>(.*)?<\/style>/g )?.[ 0 ]
```

**Pull request merged**